qcodes/dataset/sqlite/queries.py

```diff
@@ -1512,6 +1531,37 @@ def get_run_description(conn: ConnectionPlus, run_id: int) -> str:
                             "run_id", run_id)
 
 
+def get_parent_dataset_links(conn: ConnectionPlus, run_id: int) -> str:
+    """
+    Return the (JSON string) of the parent-child dataset links for the
+    specified run
+    """
+
+    # The RuntimeError comes from the atomic contextmanager as it rolls back
+    # the transaction. We assume that the rollback is caused by the column
+    # `parent_datasets` not being present. We find it fair to interpret this
+    # situation as "no parent datasets were registered"
+    #
+    # We cannot in general trust that NULLs will not appear in the column,
+    # so we must explicitly handle them.
+
```

Comment:
I would prefer to use pragma table info to check for the existence. How is a null value in the db handled here?

Reply:
Sure, we can use the pragma. It's a good idea to not just catch-all on `RuntimeError`.

A `NULL` value retrieved as a `None`. The case when `maybe_str_link is None` is handled a few lines down.

Reply:
Yes either A more specific exception catching or the pragma is fine with me